### DISCUSSION TITLE
Convert negative accounting format to numeric in python

### DISCUSSION THREAD

**Discussion by myibra** [teamalgoritma/community#281] [Q&A] [answered]
Halo semuanya, 
mau bertanya terkait mengubah format di *accounting*, jadi saya lagi ingin melakukan perhitungan pada laporan keuangan. Dimana pada laporan keuangan tersebut memiliki format tersendiri, terutama pada nilai yang negative menggunakan `()` atau *parantheses* sebagai contoh : `(1,000 k)` yang memiliki arti `1000000`.

Tetapi dalam melakukan perubahan format string tersebut terdapat kendala, sebagai contoh data yang digunakan sebagai berikut:

```
df = pd.DataFrame({'Income': ['(2,613 K)', '(54 K)', '23,688 K']})
```
<img width="95" alt="image" src="https://user-images.githubusercontent.com/50072768/187032695-141b3a28-0d13-4a49-9f4a-1bcddbf4898c.png">

code yang sudah saya coba lakukan sebagai berikut :
```
df['Income'] = df['Income'].replace('[\,)]','', regex = True).replace(' K','000', regex = True).replace('[(]','-', regex = True).astype(float)
```

dan terdapat error sebagai berikut :
`ValueError: could not convert string to float: '-'`

sekiranya kesalahan nya atau error pada bagian mana ya?

terimakasih

**Comment by IrfanChairurrachman** [comment, marked as answer]
Dear @myibra ,

Kode yang Anda ketikkan dapat bekerja dengan baik pada jupyter notebook saya, berikut saya lampirkan
![image](https://user-images.githubusercontent.com/56690035/187332379-1b46771d-375b-4547-ba4c-4f5ee96e74d5.png)

Untuk error yang Anda dapatkan, apakah terjadi ketika menggunakan selain 3 di atas? 
Error tersebut disebabkan kegagalan konversi nilai string menjadi float, kemungkinannya karena hanya string hanya bernilai "-" tanpa adanya angka yang mengikut, berikut saya sertakan lampiran kemungkinan error.
![image](https://user-images.githubusercontent.com/56690035/187332877-4954be9c-dead-49d4-95a8-9dac7275dfc1.png)

Kemungkinannya, bapak memiliki data yang kosong sehingga tidak ada nilai numerik yang dapat dikonversi menjadi float. Saran kami bapak dapat menghilangkan data kosong tersebut, atau menggunakan `try` and `except` untuk menangkap error, dan ketika error terjadi dapat menggunakan `continue` atau `pass` agar proses dapat dilanjutkan untuk data selanjutnya alih-alih program berhenti berjalan. 

Berikut kami berikan referensi untuk `try`, `except` dan `continue`.
https://www.codingem.com/python-continue-statement/

Sekian, terima kasih
